tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/ContactAutocompleteMethod.scala

```diff
@@ -0,0 +1,101 @@
+  override def jsonify(): JsObject = Json.obj()
+}
+
+case object ContactCapability extends Capability {
+  val properties: CapabilityProperties = ContactCapabilityProperties
+  val identifier: CapabilityIdentifier = LINAGORA_CONTACT
+}
+
+class ContactCapabilitiesModule extends AbstractModule {
+  @ProvidesIntoSet
+  private def capability(): CapabilityFactory = ContactCapabilityFactory
+}
+
+case object ContactCapabilityFactory extends CapabilityFactory {
+  override def create(urlPrefixes: UrlPrefixes): Capability = ContactCapability
+
+  override def id(): CapabilityIdentifier = LINAGORA_CONTACT
+}
+
+class ContactAutocompleteMethodModule extends AbstractModule {
+  override def configure(): Unit = {
+    install(new ContactCapabilitiesModule())
+    Multibinder.newSetBinder(binder(), classOf[Method])
+      .addBinding()
+      .to(classOf[ContactAutocompleteMethod])
+  }
+}
+
+class ContactAutocompleteMethod @Inject()(serializer: ContactSerializer,
+                                          emailAddressContactSearchEngine: EmailAddressContactSearchEngine,
+                                          val metricFactory: MetricFactory,
+                                          val sessionSupplier: SessionSupplier) extends MethodRequiringAccountId[ContactAutocompleteRequest] {
+  override val methodName: Invocation.MethodName = MethodName("TMailContact/autocomplete")
+  override val requiredCapabilities: Set[CapabilityIdentifier] = Set(LINAGORA_CONTACT)
+
+  override def doProcess(capabilities: Set[CapabilityIdentifier], invocation: InvocationWithContext,
+                         mailboxSession: MailboxSession, request: ContactAutocompleteRequest): Publisher[InvocationWithContext] =
+    processRequest(mailboxSession, invocation.invocation, request)
+      .map(invocationResult => InvocationWithContext(invocationResult, invocation.processingContext))
+
+  override def getRequest(mailboxSession: MailboxSession, invocation: Invocation): Either[Exception, ContactAutocompleteRequest] =
+    serializer.deserialize(invocation.arguments.value) match {
+      case JsSuccess(contactAutocompleteRequest, _) => Right(contactAutocompleteRequest)
+      case errors: JsError => Left(new IllegalArgumentException(ResponseSerializer.serialize(errors).toString))
+    }
+
+  private def processRequest(mailboxSession: MailboxSession,
+                             invocation: Invocation,
+                             request: ContactAutocompleteRequest): SMono[Invocation] = {
+    def validation: Either[Throwable, SMono[Invocation]] = for {
+      limit <- Limit.validateRequestLimit(request.limit)
+    } yield {
+      executeAutocomplete(mailboxSession, request, limit)
+        .map(response => Invocation(
+          methodName = methodName,
+          arguments = Arguments(serializer.serialize(response)),
+          methodCallId = invocation.methodCallId))
+    }
+    validation.fold(SMono.error, res => res)
```

Comment:
Limit.validateRequestLimit(request.limit).fold(...) directly ?

Reply:
Tried, can't... does not compile. Saw that we do like this in other parts of the code as well, I guess for the same reason